src/include/miopen/generic_search.hpp

```diff
@@ -466,6 +466,8 @@ auto GenericSearch(const Solver s, const Context& context_, const AnyInvokeParam
 
     MIOPEN_LOG_W("Done: " << n_runs_total << '/' << n_failed << '/' << n_runs_total << ", best #"
                           << n_best << ' ' << best_time << ' ' << best_config);
+
+    profile_h.ClearProgram();
```

Comment:
Clearing all Program objects invalidates all Kernel objects, including those that have been built _before_ tuning. This would lead to failures of applications that leverage "on-the-fly" tuning by means of `MIOPEN_FIND_ENFORCE`.

Reply:
I am working on the patch.